src/ui/utils/app.ts

```diff
@@ -0,0 +1,30 @@
+import { TimeStampedPointRange } from "@recordreplay/protocol";
+export type TimeRange = {
+  start: number;
+  end: number;
+};
+
+export function getNonLoadingRegionTimeRanges(
+  loadingRegions: TimeStampedPointRange[],
+  endTime: number
+): TimeRange[] {
+  if (!loadingRegions.length) {
```

Comment:
this was making me nervous so even though technically this is reducer-y logic I extracted it to its own util so that we can write unit tests for it

Reply:
Oooh, do we not write tests for our Redux reducers/state? 😮

Reply:
Nope 😬